### ISSUE TITLE
compiling on Raspberry Pi 3 results in undefined symbols

### ISSUE THREAD

**Issue by nwgat** [libass/libass#219]
compiled with libass 0.13.2
`symbol lookup error: /home/pi/src/mpv/build/mpv: undefined symbol: ass_set_check_readorder`
when i compiled with libass 13.0 i get 
`undefined symbol: ass_set_selective_style_override_enabled`

when playing a mkv file with ass subtitles in mpv

mpv 0.16
libass 0.13.2 or 13.0 (same in older ones too)

**Comment by nwgat** [comment]
config.log here
https://transfer.sh/lsH44/config.log

**Comment by wm4** [comment]
This has nothing to do with libass itself, and also your config.log contains something dumb (html?).

**Comment by astiob** [comment]
Uh what? They’re saying libass isn’t exporting some symbols, and the config.log is perfectly fine.

**Comment by wm4** [comment]
The symbol is exported, and it's probably a mpv-build fuckup. Not a libass one.

Trying to download config.log from that shitty site, I get a config.log filled with html tags.

**Comment by astiob** [comment]
> The symbol is exported, and it's probably a mpv-build fuckup. Not a libass one.

We don’t know yet. It’s easy to dismiss a bug report as invalid just because the bug isn’t supposed to happen—but _no_ bug is supposed to happen.

> Trying to download config.log from that shitty site, I get a config.log filled with html tags.

Well I don’t. You’re always all too quick to label anything that doesn’t immediately work for you as shitty.

**Comment by astiob** [comment]
Here’s a copy of the supplied config.log: https://gist.github.com/astiob/d99c2e732e8e2241af0d4c651017e239 Not that I see anything useful in it.

**Comment by nwgat** [comment]
@wm4 having a bad day? FYI
`curl https://transfer.sh/lsH44/config.log -o config.log`

**Comment by nwgat** [comment]
and i just compiled mpv without libass and that plays the file instead of freaking out and quits

**Comment by wm4** [comment]
This crap probably snoops the browser user agent or something equally stupid.

**Comment by nwgat** [comment]
```
Playing: ass-h264.mkv
 (+) Video --vid=1 (*) (h264)
 (+) Audio --aid=1 --alang=jpn (*) (aac)
 (+) Subs  --sid=1 --slang=eng (*) (ass)
AO: [alsa] 44100Hz stereo 2ch s16
Using hardware decoding (rpi).
VO: [rpi] 1920x1080 mmal
../src/mpv-0.16.0-with-libass/build/mpv: symbol lookup error: ../src/mpv-0.16.0-with-libass/build/mpv: undefined symbol: ass_set_selective_style_override_enabled
```

```
Compiled without libass.
There will be no OSD and no text subtitles.
Playing: ass-h264.mkv
 (+) Video --vid=1 (*) (h264)
 (+) Audio --aid=1 --alang=jpn (*) (aac)
 (+) Subs  --sid=1 --slang=eng (*) (ass)
[sub] Could not find subtitle decoder for format 'ass'.
AO: [alsa] 44100Hz stereo 2ch s16
Using hardware decoding (rpi).
VO: [rpi] 1920x1080 mmal


```

**Comment by astiob** [comment]
Hmm, that’s using static linking, isn’t it. So it seems likely indeed that the problem lies within the step compiling mpv rather than libass. Could you perhaps try to compile mpv with libass but without compiler optimizations?

And maybe, just to make sure dynamic linking works, try dynamic linking? Although is this information useful?… At the very least it seems that if dynamic linking doesn’t work either, the problem is more likely to be on libass’ side. (Or am I talking nonsense?) Oh, I guess you can just `nm libass.so` and confirm that the symbols you’re currently missing are in fact exported—rather than actually recompile mpv with dynamic linking against libass.

**Comment by astiob** [comment]
I’ve just built mpv with a statically linked libass using [mpv-build](https://github.com/mpv-player/mpv-build) (current master: mpv-player/mpv-build@a27a1e462d3650ab691e47568f07b942c37f0519) on a clean Raspbian install on Raspberry Pi 3, and ASS subtitles work with no problems.

I did `sudo apt-get install autoconf libtool libfribidi-dev libfontconfig-dev` followed by `./rebuild -j4`. I then tested the resulting binary and saw no issues. Just to be sure this isn’t related to HarfBuzz (which I see in your config.log), I then did `sudo apt-get install libharfbuzz-dev` and another `./build -j4` and tested the new binary: subtitles still worked.

[Logs](https://gist.github.com/astiob/da6fee0eb4ac77a862d682107b267a6c)
